wrappers/csharp/cs-tutorial-8-D400-on-chip-calibration/ExampleAutocalibrateDevice.cs

```diff
@@ -0,0 +1,541 @@
+            //row.Add($@"{timeout:mm\:ss\.fff}"); 
+            row.Add($@"{calibrationTakes:mm\:ss\.fff}");
+            row.Add($"{deviceHealth}");
+            row.Add($"{deviceHealthDescription}");
+            row.Add($"{comments}");
+
+            return string.Join(",", row);
+        }
+        private static bool IsLinux
+        {
+            get
+            {
+                int p = (int)Environment.OSVersion.Platform;
+                return (p == 4) || (p == 6) || (p == 128);
+            }
+        }
+        private static int CountEnumEntries<T>() where T : struct, IConvertible
+        {
+            if (!typeof(T).IsEnum)
+                throw new ArgumentException("T must be an enumerated type");
+
+            return Enum.GetNames(typeof(T)).Length;
+        }
+        private static T[] GetEnumEntries<T>() where T : struct, IConvertible
+        {
+            if (!typeof(T).IsEnum)
+                throw new ArgumentException("T must be an enumerated type");
+
+            return (T[])Enum.GetValues(typeof(T));
+        }
+
+        #endregion CalibrationModesTest
+
+        private Pipeline StartPipeline(VideoStreamProfile depthProfile, Device dev)
+        {
+            var cfg = new Config();
+            cfg.EnableDevice(dev.Info[CameraInfo.SerialNumber]);
+            cfg.EnableStream(Stream.Depth, depthProfile.Width, depthProfile.Height, depthProfile.Format, depthProfile.Framerate);
+
+            var pipeline = new Pipeline();
+            var pp = pipeline.Start(cfg);
+
+            return pipeline;
+        }
+        private DeviceHealth GetDeviceHealth(double health)
+        {
+            health = System.Math.Abs(health);
+            if (health < 0.25)
+                return DeviceHealth.Good;
+            else if (health < 0.75)
+                return DeviceHealth.CouldBePerformed;
+            else
+                return DeviceHealth.NeedRecalibration;
+        }
+        private string GetCalibrationConfig(CalibrationSpeed calibrationSpeed, CalibrationScanParameter calibrationScanParameter, CalibrationDataSampling calibrationDataSampling)
+        =>
+                        $@"{{
+	                        ""speed"": {(int)calibrationSpeed},
+	                        ""scan parameter"": {(int)calibrationScanParameter},
+	                        ""data sampling"": {(int)calibrationDataSampling}
+                        }}";
+
+        public static ConsoleKey ConsoleGetKey(ConsoleKey[] possibleKeys, string prompt)
+        {
+            ConsoleKey res = ConsoleKey.Enter;
+            do
+            {
+                Console.WriteLine(prompt);
+                res = Console.ReadKey(true).Key;
+            }
+            while (!possibleKeys.Contains(res));
+
+            return res;
+        }
+        public static bool IsTheDeviceD400Series(Device dev)
+        {
+            var res = Regex.IsMatch(dev.Info[CameraInfo.Name].ToString(), @"(D4\d\d)");
```

Comment:
device may be null - even in the flow currently used

Reply:
done